From dcd7ebf37b0333191c9be07cd7e8f6a5fdadd567 Mon Sep 17 00:00:00 2001
From: Chris Wedgwood <cw@f00f.org>
Date: Thu, 7 Mar 2019 21:45:32 +0000
Subject: [PATCH] [neutron] 'up' interfaces added via ovs auto_bridge_add

Previously, when adding interfaces to an ovs bridge we would set the
link state to up.  Some environments assume this is the case so
restore that behavior.

This fixes the problem where external (public) IPs for routers and VMs
no longer respond.

Change-Id: I59e21bd5cde7e239320125e9a7e0a33adae578a8
---
 neutron/templates/bin/_neutron-openvswitch-agent-init.sh.tpl | 1 +
 1 file changed, 1 insertion(+)

diff --git a/neutron/templates/bin/_neutron-openvswitch-agent-init.sh.tpl b/neutron/templates/bin/_neutron-openvswitch-agent-init.sh.tpl
index 84f5e4b..4dfb0ff 100644
--- a/neutron/templates/bin/_neutron-openvswitch-agent-init.sh.tpl
+++ b/neutron/templates/bin/_neutron-openvswitch-agent-init.sh.tpl
@@ -45,6 +45,7 @@ do
   if [ -n "$iface" ] && [ "$iface" != "null" ]
   then
     ovs-vsctl --no-wait --may-exist add-port $bridge $iface
+    ip link set dev $iface up
   fi
 done
 
-- 
2.7.4

